pycontrails/models/ps_model/ps_model.py

```diff
@@ -0,0 +1,936 @@
+"""Simulate aircraft performance using Poll-Schumann (PS) model.
+
+References
+----------
+Poll & Schumann (2021). An estimation method for the fuel burn and other performance characteristics of civil
+    transport aircraft in the cruise. Part 1: fundamental quantities and governing relations for a general atmosphere.
+    Aero. J., 125(1284), 296-340, doi: 10.1017/aer.2020.62.
+
+Poll & Schumann (2021a). An estimation method for the fuel burn and other performance characteristics of civil
+    transport aircraft during cruise: Part 2, determining the aircraft’s characteristic parameters. Aero. J.,
+    125(1284), 257-295, doi: 10.1017/aer.2020.124.
+
+Poll & Schumann (2022). An estimation method for the fuel burn and other performance characteristics of civil
+    transport aircraft. Part 3 Generalisation to cover climb, descent and holding. Aero. J., submitted.
+"""
+
+from __future__ import annotations
+
+import pathlib
+import warnings
+import typing
+import dataclasses
+import numpy as np
+import numpy.typing as npt
+from typing import Mapping, Any
+
+from pycontrails.core.models import Model, ModelParams
+from pycontrails.core import flight
+from pycontrails.core.fuel import Fuel, JetA
+from pycontrails.physics import constants, jet, units
+from pycontrails.models.aircraft_performance import AircraftPerformanceData
+from pycontrails.models.ps_model.aircraft_params import AircraftEngineParams, get_aircraft_engine_params
+
+_path_to_static = pathlib.Path(__file__).parent / "static"
+
+default_path: str | pathlib.Path = _path_to_static / "ps-aircraft-params-20230427.csv"
+
+
+# TODO: To link up PSModelParams in review
+
+@dataclasses.dataclass
+class PSModelParams(ModelParams):
+    """:class:`PSModel` model parameters."""
+
+    #: Default paths
+    aircraft_database_path: str | pathlib.Path = _path_to_static / "ps-aircraft-params-20230425.csv"
+
+    #: Fuel type
+    fuel: Fuel = dataclasses.field(default_factory=JetA)
+
+
+class PSModel:
+    name = "PS Model"
+    long_name = "Poll-Schumann Aircraft Performance Model"
+    default_params = PSModelParams
+
+    aircraft_engine_params: Mapping[str, AircraftEngineParams]
+
+    def __init__(
+            self,
+    ) -> None:
+
+        # Set class variable with engine parameters if not yet loaded
+        if not hasattr(self, "aircraft_engine_params"):
+            type(self).aircraft_engine_params = get_aircraft_engine_params(default_path)
```

Comment:
If you go for the `functools.cache` approach I mentioned earlier, this could be a little more natural:

```
self.aircraft_engine_params = get_aircraft_engine_params(default_path)
```

Should `default_path` here be replaced by `self.params["aircraft_database_path"]`?